pacehrh/R/pace_coverage_config.R

```diff
@@ -0,0 +1,78 @@
+#' Load CoverageRates Values
+#'
+#' Read the annual task coverage values from the model inputs Excel file. The
+#' name and location of the model inputs Excel file is loaded from the global
+#' configuration JSON file.
+#'
+#' @param sheetName Sheet name from the model input Excel file
+#'
+#' @return Data table of per-year task coverage levels
+#'
+loadCoverageRates <- function(sheetName = .defaultCoverageRatesSheet){
+  traceMessage(paste0("Loading task coverage rates sheet ", sheetName))
+  
+  coverageRatesData <- NULL
+  
+  coverageRatesData <- loadTable(sheet = sheetName) # loadTable does validation
+  
+  hr <- colnames(coverageRatesData)
+  
+  # Sanity check: first two columns should be indicator and common name:
+  if (!identical(hr[1:2], c("Indicator", "CommonName"))) {
+    traceMessage(
+      paste0(
+        "Incorrect coverage rate config header names: (",
+        as.character(hr),
+        ")"
+      )
+    )
+    return(NULL)
+  }
```

Comment:
This code generates an ugly error message when the CommonName label is changed to CommonName_xxx:

Loading task coverage rates sheet Coverage
Incorrect coverage rate config header names: (Indicator)Incorrect coverage rate config header names: (CommonName_xxx)Incorrect coverage rate config header names: (Year 0)Incorrect coverage rate config header names: (Year 1)Incorrect coverage rate config header names: (Year 2)Incorrect coverage rate config header names: (Year 3)Incorrect coverage rate config header names: (Year 4)Incorrect coverage rate config header names: (Year 5)Incorrect coverage rate config header names: (Year 6)Incorrect coverage rate config header names: (Year 7)Incorrect coverage rate config header names: (Year 8)Incorrect coverage rate config header names: (Year 9)Incorrect coverage rate config header names: (Year 10)Incorrect coverage rate config header names: (Year 11)Incorrect coverage rate config header names: (Year 12)Incorrect coverage rate config header names: (Year 13)Incorrect coverage rate config header names: (Year 14)Incorrect coverage rate config header names: (Year 15)Incorrect coverage rate config header names: (Year 16)Incorrect coverage rate config header names: (Year 17)Incorrect coverage rate config header names: (Year 18)Incorrect coverage rate config header names: (Year 19)Incorrect coverage rate config header names: (Year 20)